src/datatypes.ts

```diff
@@ -153,6 +156,75 @@ export function deserializeDatetime(literalString: string): Date | null {
   return date;
 }
 
+/**
+ * @internal
+ * @param value Value to serialise.
+ * @returns String representation of `value`.
+ * @see https://www.w3.org/TR/xmlschema-2/#date-lexical-representation
+ */
+export function serializeDate(value: Date): string {
+  // Although the XML Schema Date is not _exactly_ an ISO 8601 string
+  // (see https://www.w3.org/TR/xmlschema-2/#deviantformats),
+  // the deviations only affect the parsing, not the serialisation.
+  // Therefore, we can just use .toISOString():
+  return value.toISOString();
+}
+
+/**
+ * @internal
+ * @param value Value to deserialise.
+ * @returns Deserialized datetime, or null if the given value is not a valid serialised datetime.
+ * @see https://www.w3.org/TR/xmlschema-2/#date-lexical-representation
+ */
+export function deserializeDate(literalString: string): Date | null {
+  // Date in the format described at
+  // https://www.w3.org/TR/xmlschema-2/#date-lexical-representation
+  // (without constraints on the value).
+  // -? - An optional leading `-`.
+  // \d{4,}- - Four or more digits followed by a `-` representing the year. Example: "3000-".
+  // \d\d-\d\d - Two digits representing the month and two representing the day of the month,
+  //             separated by a `-`. Example: "11-03".
```

Comment:
Here, too, I think it's missing the fact that it can optionally also contain a timezone (i.e. `Z`, or `+01:30` or `-02:00`).